src/Database/PostgreSQL/Transaction.hs

```diff
@@ -115,14 +121,19 @@ executeOne :: (ToRow input, MonadIO m)
            => input
            -> Postgres.Query
            -> PGTransactionT m Bool
-executeOne params q = do
-  results <- execute q params
-  return (results == 1)
+executeOne params q = (== 1) <$> execute params q
+
+-- | Lookup a single FromField value. This takes care of handling 'Only' for you.
+queryOnly :: (ToRow params, FromField f, MonadIO m)
+          => params
+          -> Postgres.Query
+          -> PGTransactionT m (Maybe f)
+queryOnly params q = fmap Postgres.fromOnly <$> queryHead params q
 
 formatQuery :: (ToRow q, MonadIO m)
-            => Postgres.Query
-            -> q
+            => q
```

Comment:
Kinda confusing that `q` is the type parameter but `query` is the bound parameter